### PR TITLE
Add support for child loggers

### PR DESCRIPTION
**Pull request by nicokaiser**
Some Log4j compatible logger like Bunyan or Pino support child loggers: https://github.com/pinojs/pino/blob/HEAD/docs/child-loggers.md 

This PR adds a `child()` methods that returns another logger. This enables `abstract-logging` to mock loggers in existing projects (which use child loggers).

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -15,3 +15,10 @@ assert.equal(result1, undefined)
 
 const result2 = two.info()
 assert.equal(result2, 'info')
+
+assert.ok(one.child)
+assert.equal(Function.prototype.isPrototypeOf(one.child), true)
+
+const three = two.child()
+assert.ok(three.info)
+assert.equal(Function.prototype.isPrototypeOf(three.info), true)
```
Comment (jsumners) [inline test.js:24]: Testing for a grandchild by creating a `four` instance from the `three` instance would be a good idea.

**Review by jsumners — COMMENTED** [review]
Give me some time to think about this. The idea behind this module is to provide the minimal interface. As you are aware, adding this feature is easily done by users:

https://github.com/fastify/fastify/blob/797a9db922e67c0521f8e09d68305aacfafc02d7/lib/logger.js#L79-L82

However, this module _is_ part of the JavaScript ecosystem and you are correct, the `.child` method is prevalent within it.

**Comment by jsumners** [comment]
@nicokaiser instead of this, would you be opposed to a documentation update that shows how to extend the object to add a null child logger method?

**Comment by nicokaiser** [comment]
As the Log4j interface does not really have the `child` method, it would be ok to mention in the docs how to extend it, you are right.

**Comment by jsumners** [comment]
> As the Log4j interface does not really have the `child` method, it would be ok to mention in the docs how to extend it, you are right.

Excellent. Would you like to create such a PR? We can close this one for posterity.

**Pull request closed without merging**